vamb/__main__.py

```diff
@@ -1248,37 +1299,32 @@ def write_clusters_and_bins(
     clusters: dict[str, set[str]],
```

Comment:
Consider making this argument `Iterable[tuple[str, set[str]]]`. As far as I cal tell, we don't need an actual dict here. Adjust the calls to `.items()` in this function accordingly. Then, you can avoid creating the single element dict in `write_clusters_table` when calling this function